src/main.rs

```diff
@@ -109,6 +122,71 @@ fn main_with_args(opts: Args) -> Result<(), Box<dyn Error>> {
             .build_global()?;
     }
 
+    // Construct the status update channel
+    let (status_tx, status_rx): (
+        std::sync::mpsc::Sender<Message>,
+        std::sync::mpsc::Receiver<Message>,
+    ) = std::sync::mpsc::channel();
+
+    // Initialize the logger if not writing to a file, otherwise we defer it to after we created the MultiProcess
+    let mut log_builder = env_logger::Builder::new();
+    log_builder.filter_level(opts.verbose.log_level_filter());
+    if !to_file {
+        // Initialize the logger with the provided verbosity
+        let _ = log_builder.try_init();
+    }
+
+    // If we are outputing to a file and not stdout, start a gui thread that uses indicatif to display progress
+    // If we would always start a MultiProgress progress bar, the output would be mangled by the output we write to stdout ourselves.
+    let handle = if to_file {
+        let spinner_style =
+            indicatif::ProgressStyle::with_template("{prefix:.bold.dim} {spinner} {wide_msg}")?
+                .tick_chars("⠁⠂⠄⡀⢀⠠⠐⠈ ");
+
+        // Create the MultiProgress
+        let multi = indicatif::MultiProgress::new();
+
+        let logger = log_builder.build();
+
+        // Initialize the logger
+        let _ = indicatif_log_bridge::LogWrapper::new(multi.clone(), logger).try_init();
+
+        Some(std::thread::spawn(move || {
+            // Create a progress bar for rayon thread in the global thread pool
+            let mut progress_bars = Vec::new();
+            for _ in 0..rayon::current_num_threads() {
+                let pb = multi.add(indicatif::ProgressBar::new(0));
+                pb.set_style(spinner_style.clone());
+                progress_bars.push(pb);
+            }
+
+            for message in status_rx {
+                match message.status {
+                    nixtract::message::Status::Started => {
+                        progress_bars[message.id]
```

Comment:
They are always consecutive, and since we only use the global thread pool, they are indeed between 0 and `current_num_threads()`. In fact, `rayon` calls them indexes, which is indeed a better name for them.